aws/resource_aws_iam_role.go

```diff
@@ -311,34 +313,41 @@ func resourceAwsIamRoleUpdate(d *schema.ResourceData, meta interface{}) error {
 }
 
 func resourceAwsIamRoleDelete(d *schema.ResourceData, meta interface{}) error {
-	iamconn := meta.(*AWSClient).iamconn
+	conn := meta.(*AWSClient).iamconn
 
-	// Roles cannot be destroyed when attached to an existing Instance Profile
```

Comment:
maybe keep the removed comments so we don't lose the rational if this Delete function needs to be modified in the future?

Reply:
Good point